### PR TITLE
Update proto files, sample code and unit tests.

### PR DESCRIPTION
**Pull request by scalarindetail**
This PR implements #33 and #34

### CODE REVIEW

File: package.json
Comment (scalarindetail) [inline package.json:13]: Waiting for the base to publish a newer version.

**Review by Torch3333 — COMMENTED** [review]
It looks good. Let's wait for the base to be publish to be able to run the CI

File: .circleci/config.yml
Comment (scalarindetail) [inline .circleci/config.yml:69]: Sorry i couldn't think of a good wait to wait for the container to be ready.

Comment (Torch3333) [reply .circleci/config.yml:69]: Without the sleep, the test fails I guess ?

Comment (scalarindetail) [reply .circleci/config.yml:59]: Yes, i will fail due to the container being able to be ready on time.

File: .circleci/config.yml
Comment (Torch3333) [inline .circleci/config.yml:45]: It was not working with Node `14.16` ?

Comment (scalarindetail) [reply .circleci/config.yml:45]: Sorry i haven't try `14.16`

File: .circleci/config.yml
Comment (feeblefakie) [inline .circleci/config.yml:8]: What is the main reason of using a machine and install docker?
For using docker-compose?

File: .circleci/config.yml
Comment (feeblefakie) [inline .circleci/config.yml:24]: Are you currently using your account for this?

File: .circleci/docker-compose.yml
Comment (feeblefakie) [inline .circleci/docker-compose.yml:1]: Is it a copy from `scalar-samples`?
If so, I think it would be better to clone the repo and reuse it for maintainability.

File: .circleci/envoy.yaml
Comment (feeblefakie) [inline .circleci/envoy.yaml:1]: ditto.

File: scalardl-node-client-sdk.js
Comment (feeblefakie) [inline scalardl-node-client-sdk.js:42]: IMHO, the following structure looks slightly better even though more nested.
```
if (tlsEnabled) {
    if (ca) {

    } else (

    }
} else {

}
```

File: .circleci/config.yml
Comment (feeblefakie) [inline .circleci/config.yml:29]: Just a question. Why not using the apt package?
I think it should be noted if there is a particular reason.

File: .circleci/config.yml
Comment (feeblefakie) [reply .circleci/config.yml:69]: Waiting what container to be ready?

**Review by feeblefakie — COMMENTED** [review]
Overall looking good! Thank you!
Left some comments, suggestions, and questions.

File: .circleci/config.yml
Comment (scalarindetail) [reply .circleci/config.yml:8]: Yes, for using docker-compose, also in circleci, if we use docker executor, the docker container that got spun up is remote, [so we can't use volume](https://support.circleci.com/hc/en-us/articles/360007324514-How-can-I-use-Docker-volume-mounting-on-CircleCI-), we need it to load our function table schema `foo.bar` into the cassandra.

File: .circleci/config.yml
Comment (scalarindetail) [reply .circleci/config.yml:24]: Yes, im using the scalarindetail account for this

File: .circleci/config.yml
Comment (scalarindetail) [reply .circleci/config.yml:29]: The [docs here](https://circleci.com/docs/2.0/configuration-reference/#available-machine-images) is stating the latest version, but after i debug it (ssh into the session), I found out that the docker version the image is using is a very old version and only supported up to docker-compose file of v3.2 which is not as they stated on that docs, so i have to manually update the docker version.

**Review by jnmt — APPROVED** [review]
LGTM! Left one minor suggestion. Thank you!

File: scalardl-node-client-sdk.js
Comment (Torch3333) [reply scalardl-node-client-sdk.js:42]: Sure, I will update it.

File: scalardl-node-client-sdk.js
Comment (jnmt) [inline scalardl-node-client-sdk.js:51]: A single line is better if we can keep it less than 80 chars (and there is no policy for doing this).

File: scalardl-node-client-sdk.js
Comment (jnmt) [inline scalardl-node-client-sdk.js:55]: Same as above.

**Review by jnmt — COMMENTED** [review]
Just a minor comment. Thank you :-)

File: .circleci/config.yml
Comment (Torch3333) [inline .circleci/config.yml:36]: Could you update this text to explains why you are checking out this repository as it is probably not obvious ?

File: .circleci/config.yml
Comment (Torch3333) [inline .circleci/config.yml:63]: What do you think ?
```suggestion
          name: Wait for Cassandra to spin up to be able to update the Cassandra schema in the next step
```

File: .circleci/config.yml
Comment (Torch3333) [inline .circleci/config.yml:44]: This maybe is more explanatory.
```suggestion
          name: Start Scalar DL environment 
```

**Review by Torch3333 — COMMENTED** [review]
Thanks, I left some suggestions for the CI steps to be more explanatory.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Comment by feeblefakie** [comment]
@jnmt I'll merge it but PTAL when you get a chance 🙇

**Comment by jnmt** [comment]
@feeblefakie LGTM and just worried about the same point as you mentioned.
https://github.com/scalar-labs/scalardl-node-client-sdk/pull/35#discussion_r591166051
@Torch3333 Please make sure to also update the list of environment variables in README_CI.md if necessary.
Nice work. Thank you!

**Pull request merged**
